internal/apps/dop/services/pipeline/pipeline.go

```diff
@@ -424,6 +430,34 @@ func (p *Pipeline) ConvertPipelineToV2(pv1 *apistructs.PipelineCreateRequest) (*
 	return pv2, nil
 }
 
+func (p *Pipeline) getPipelineOwnerUser(app *apistructs.ApplicationDTO, pv1 *apistructs.PipelineCreateRequest) (*apistructs.PipelineUser, error) {
+	ymlCommit, err := p.bdl.GetGittarTree(fmt.Sprintf("/%s/tree/%s/%s", app.GitRepoAbbrev, pv1.Branch, pv1.PipelineYmlName), strconv.FormatUint(app.OrgID, 10), pv1.UserID)
+	if err != nil {
+		return nil, err
+	}
+	if ymlCommit.Commit.Committer == nil || ymlCommit.Commit.Committer.Name == "" {
+		return nil, fmt.Errorf("pipeline yml commit committer is empty")
+	}
+	authorEmail := ymlCommit.Commit.Committer.Email
+	users, err := p.bdl.ListUsers(apistructs.UserListRequest{
+		Query:     authorEmail,
```

Comment:
any exact-match method? otherwise your performance will be terrible.

Reply:
non exact-match method, the current performance is acceptable